test_address_book.py

```diff
@@ -0,0 +1,40 @@
+import unittest
+import address_book
+from pathlib import Path
+
+
+class TestAddressBook(unittest.TestCase):
+    def setUp(self):
+        self.john = address_book.Contact("John", "email.address", "123")
+        self.ab = address_book.AddressBook('contact.json')
+        self.ab.add_contact('John', self.john)
+
+    def tearDown(self):
+        Path(self.ab.contactsfile).unlink()
+
+    def test_show(self):
+        self.assertEqual(self.john.name, "John")
+        self.assertEqual(self.john.email_address, "email.address")
+        self.assertEqual(self.john.phone_number, "123")
+
+    def test_add_contact(self):
+        self.assertIn('John', self.ab._contacts)
```

Comment:
you should actually add a new contact here and check that it is in address book. eg add one called jack